### ISSUE TITLE
Wire Library endTransmission return forever 0

### ISSUE THREAD

**Issue by rromano001** [energia/Energia#617]
Linux Mint 17.2 AMD64
Intel Core I7
Energia 14 - Energia 15
Target Launchpad G msp430G2553 (same for MSP430G2332)

 I was trying to import lcd libraries i2c based, I found I2C library is not working. After some reading I wrote a code similar to scenner I late found, if I write a byte to I2C library hang, if I don't wrote byte endTransmission forever return 0 also if SCL is stuck at 0.
 endTransmission default true or false? No mention on documentation or I was not able to find. begin-end trnsmission without a write in the middle start address transmission or some other setting are to be passed? 

here result and code I used:

>  no write:
> device 0x3C Errnum: 0
> device 0x3D Errnum: 0
> device 0x3E Errnum: 0
> device 0x3F Errnum: 0
> device 0x40 Errnum: 0
>  Wire.write in the middle:
> device 0x01 Errnum: 3
> device 0x02 Errnum: 3
> device 0x03 Errnum: 3
> device 0x04Scan Begin
> device 0x01

 a very long time pass before Errnum is printed.

Code used:

```
// I2C Scanner
/* start from address 0 to 0x7f of I2C searching devices.
*/

#include <Wire.h>

void setup()
{
  Serial.begin(9600);  // start serial for output
  Serial.println("Scan Begin");
  Wire.begin();        // join i2c bus (address optional for master)
}

int icaddr = 1;
int errnum;

void loop()
{
  Serial.println();
  Serial.print("device 0x");
  if (icaddr<16) Serial.print(0);
  Serial.print(icaddr,HEX);
  Wire.beginTransmission(icaddr++);    // request 6 bytes from slave device #2
  Wire.write(1);
  errnum=Wire.endTransmission();
  delay(5);
  Serial.print(" Errnum: ");
  Serial.print(errnum);
  if(errnum) delay(2000);
  if (icaddr>0x7f)
    icaddr=1;
}
```

**Comment by rromano001** [comment]
After reading code, Wire.endTransmission() default to (stop), transmission is initiated only if a payload byte is present so no Wire.write() no transmission at all.

 First attempt of debug was an PCF 8574(a) based LCD, Library using this device was prepared for a different pinout of LCD. After scanning bit of PCF I found original issue on target board.

 Debug issue was generated on Launchpad, target board with pullup where OK, Launchpad has LED on P1.6, attaching Logic analyser to these two pin reported I2C as correctly working, attaching a scope I see P1.6 waveform was weak and taking an eye on LP I seen led glowing, so 1.9V are ok for LA set to 1.8 threshold but not for MSP I2C interface.

 Remove LED P1.6 Jumper cure this strange problem, more in deep analysis can be done when I get a new LP5529 to debug library on CCS. (LP G series is not supported on Linux.)

 A timeout on library is missing so IMHO it must be added to terminate unresponsive bus or stuck peripheral.

 Documentation site, no mention of default I2C channel on TIVA LP or how to select different channel.

*******************************************************************************
**  Try remove LED P1.6 Jumper cure this strange problem   **
*******************************************************************************

**Comment by rromano001** [comment]
Hi JRDavis, sorry to say I never used nothing related to TI product.
 I got too much attack from user on B2B so I moved to another uC and also removed every TI product from my board.
 MSP430 was a great processor, TI never understood that so I found a new light over old sky.
 Again I can't teach everyone has no knowledge of basic C programming for free.
 Vilo work was also great but need support from provider so i can only say  Sorry I close this and I detach from.
 Best regards.
 Roberto
